packages/clerk-js/src/ui/components/ApiKeys/utils.ts

```diff
@@ -0,0 +1,60 @@
+import { useCallback, useEffect, useRef } from 'react';
+import { useSWRConfig } from 'swr';
+
+type UseApiKeysPaginationParams = {
+  query: string;
+  page: number;
+  pageCount: number;
+  isFetching: boolean;
+  subject: string;
+  fetchPage: (page: number) => void;
+};
+
+/**
+ * Hook that manages pagination logic for API keys:
+ * - Resets to page 1 when query changes
+ * - Adjusts page when current page exceeds available pages (e.g., after deletion)
+ * - Provides cache invalidation function for mutations
+ */
+export const useAPIKeysPagination = ({
```

Comment:
🙃 Let's look into consistency between `API` & `Api`. It seems they're used interchangeably and it'll be good to settle on one.